release-notes/4.0/4.0.0-preview1.md

```diff
@@ -0,0 +1,122 @@
+- Fixed `EntryPointNotFoundException` in `InOutOfProcHelper` constructor. [#1120](https://github.com/dotnet/SqlClient/pull/1120)
+- Fixed async thread blocking issues on `SqlConnection.Open()` for active directory authentication modes. [#1213](https://github.com/dotnet/SqlClient/pull/1213)
+- Fixed driver behavior for always encrypted with secure enclaves to not fail when no user parameters have been provided.[#1115](https://github.com/dotnet/SqlClient/pull/1115)
+- Fixed bug with `LegacyRowVersionNullBehavior` App Context switch. [#1182](https://github.com/dotnet/SqlClient/pull/1182)
+- Fixed issues in Strings.resx file containing error messages. [#1136](https://github.com/dotnet/SqlClient/pull/1136)
+ and [#1178](https://github.com/dotnet/SqlClient/pull/1178)
+
+### Changed
+- Updated error code to match with Windows when certificate validation fails in non-Windows client environments. [#1130](https://github.com/dotnet/SqlClient/pull/1130)
+- Removed designer attributes from `SqlCommand` and `SqlDataAdapter`. [#1132](https://github.com/dotnet/SqlClient/pull/1132)
+- Updated configurable retry logic default retriable error list. [#1125](https://github.com/dotnet/SqlClient/pull/1125)
+- Improved performance by changing `SqlParameter` bool fields to flags. [#1064](https://github.com/dotnet/SqlClient/pull/1064)
+- Improved performance by implementing static delegates. [#1060](https://github.com/dotnet/SqlClient/pull/1060)
+- Optimized async method allocations in .NET Framework by porting changes from .NET Core. [#1084](https://github.com/dotnet/SqlClient/pull/1084)
+
+## New features over stable release v3.0
+
+### Encrypt default value set to true
+
+The default value of the `Encrypt` connection setting has been changed from `false` to `true`. With the growing use of cloud databases and the need to ensure those connections are secure, it's time for this backwards-compatibility-breaking change.
+
+### Ensure connections fail when encryption is required
+In scenarios where client encryption libraries were disabled or unavailable, it was possible for unencrypted connections to be made when Encrypt was set to true or the server required encryption.
+
+###  App Context Switch for using System default protocols
+
+TLS 1.3 is not supported by the driver; therefore, it has been removed from the supported protocols list by default. Users can switch back to forcing use of Operating System's client protocols, by enabling the App Context switch below:
+
+ `Switch.Microsoft.Data.SqlClient.UseSystemDefaultSecureProtocols`
+
+### Enable optimized parameter binding
+
+The new API takes two Boolean values of `true` or `false`.
+When set to true parameters names will not be sent to the SQL server when the command is executed. This behavior has been shown to increase performance for commands with very large numbers of parameters which are used in bulk operations.
```

Comment:
```suggestion
When set to true parameter names will not be sent to the SQL server when the command is executed. This behavior has been shown to increase performance for commands with very large numbers of parameters which are used in bulk operations. This also means that you can't reference parameters by name when using this option.
```

Reply:
You can use the names. You just don't get any output parameters sent back to you. The names are sent in two places, the query and the parameter data and in this case we omit the name in the parameter data section. The effect of this is as if a lookup list of name=value mappings isn't created and this means that the list isn't constantly traversed to find names (which is where the perf increase comes from) but also that when you want to write the output value you can't find it by name to do so and so you don't get output values sent back.

Reply:
done.

Reply:
@Wraith2 Can you put it in 1 sentence that we should update it here? :)

Reply:
David's changes are fine without the last sentence that was added.